alf/config_helpers.py

```diff
@@ -134,6 +168,10 @@ def parse_config(conf_file, conf_params):
     finally:
         _is_parsing = False
 
+    # TODO(breakds): Decouple this with configuration parsing after get_env is
+    # decoupled from parse_config.
+    adjust_config_by_multi_process_divider(multi_process_divider)
```

Comment:
It might be too late to do this here. get_env() might already be called by the config file (e.g. though get_observation_spec()).

Reply:
Thanks. Yes `get_observationjspec()` can be called from `*_conf.py`. However I feel reluctant about moving it further to `get_env` because unlike `parse_conf`, `get_env()` is not a "once per process" function and we probably want to keep its argument clean.

How does the following idea appeal to you? We could add a global variable context that stores the rank and number of processes (read only), so that all functions can get access to it without having to pass them down along the stack. I hate globals too but these read-only global variables might help reduce the damage to readability.

Reply:
That sounds a reasonable solution to me.